src/pyscipopt/scip.pyx

```diff
@@ -647,21 +647,33 @@ cdef class Model:
     cdef public object data
     # make Model weak referentiable
     cdef object __weakref__
+    # flag to indicate whether the SCIP should be freed. It will not be freed if an empty Model was created to wrap a
+    # C-API SCIP instance.
+    cdef SCIP_Bool freescip
 
-    def __init__(self, problemName='model', defaultPlugins=True, Model sourceModel=None, origcopy=False, globalcopy=True, enablepricing=False):
+    def __init__(self, problemName='model', defaultPlugins=True, Model sourceModel=None, origcopy=False, globalcopy=True, enablepricing=False, noscipcreate=False):
         """
         :param problemName: name of the problem (default 'model')
         :param defaultPlugins: use default plugins? (default True)
         :param sourceModel: create a copy of the given Model instance (default None)
         :param origcopy: whether to call copy or copyOrig (default False)
         :param globalcopy: whether to create a global or a local copy (default True)
         :param enablepricing: whether to enable pricing in copy (default False)
+        :param noscipcreate: initialize the Model object without creating a SCIP instance
```

Comment:
I don't like the double negative. Why not flip it and call it `createscip`?

Reply:
If we have an additional parameter, such as `sourcescip`, then we could could eliminate this parameter. The logic would be if `sourcescip is not None`, then a model is created with the sourcescip as the scip instance.

Does that work for you?

Reply:
Yes, that's what I had in mind. Similar to how `sourceModel` works. But then the other parameters `origcopy`, `globalcopy` don't make much sense, right...?

Reply:
No the `origcopy` and `globalcopy` don't make sense because this is not a copy of SCIP.

Reply:
There is a problem with adding a SCIP* as an optional argument to the `__init__` function. SCIP* is a C variable, where the `__init__` function can only take python variables. So, it is not possible to initialise an empty Model with an available SCIP*.

We need to go with a static method that can insert the SCIP* into an empty model.